pkg/resourcegenerator/resourceutils/helpers.go

```diff
@@ -18,24 +29,41 @@ func ShouldScaleToZero(jsonReplicas *apiextensionsv1.JSON) bool {
 	return false
 }
 
-func GetImageVersion(imageVersionString string) string {
-	parts := strings.Split(imageVersionString, ":")
+// HumanReadableVersion returns the version part of an image string
+func HumanReadableVersion(imageReference string) string {
```

Comment:
I feel like this got too complicated

Personally I'd prefer it to just be simple, or not set it at all... something like this: 

```
versionPart := strings.Split(imageReference)[0]
version := versionPart[:LabelValueMaxLength]
if allowedChars.MatchString(version) {
  return version
}
return "" (or error) 
```